### ISSUE TITLE
Investigate service-soak failure

### ISSUE THREAD

**Issue by jayunit100** [kubernetes/kubernetes#15629]
Last night,  the service soak test (#9052) was found to create a regression and reverted.  The original issue is https://github.com/kubernetes/kubernetes/issues/15599.

- It works locally.
- E2E's pass in the CI.  Possibly they skip this test.
- @dchen1107 might know: Was this a stochastic failure?  I see "kind/flake" label on the issue.  Or was it 100% failure  ...
- **Possibility 1**: It appears that networking.go is labelled as flakey in https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/e2e.sh .  This tests makes **that label a no-op** because it renames it....  I wonder if, given that the flakey networking we remove that flakey label?
- **Possibility 2**: The stack trace seems to indicate its a `BeforeEach` failure happening in the setup phase, which is also quite interesting.  Maybe some collision of test resources.  
- Other possibilities?

For consolidating info: Here is the copy that was mentioned in the original issue, plus the logs...
```
Kubernetes e2e suite.EmptyDir volumes should support (root,0666,default) [Conformance]
Kubernetes e2e suite.Services should serve multiport endpoints from pods [Conformance]

Both tests are panic with a similar stacktrace. 

19:52:11 •! Panic in Spec Setup (BeforeEach) [7.044 seconds]
19:52:11 EmptyDir volumes
19:52:11 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/empty_dir.go:95
19:52:11   should support (root,0666,default) [Conformance] [BeforeEach]
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/empty_dir.go:78
19:52:11 
19:52:11   Test Panicked
19:52:11   runtime error: invalid memory address or nil pointer dereference
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:401
19:52:11 
19:52:11   Full Stack Trace
19:52:11   /usr/src/go/src/runtime/panic.go:42 (0x41452e)
19:52:11   /usr/src/go/src/runtime/sigpanic_unix.go:26 (0x41ace4)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:270 (0x511e50)
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:401 (0x43e715)
19:52:11   /usr/src/go/src/runtime/panic.go:387 (0x415388)
19:52:11   /usr/src/go/src/runtime/panic.go:42 (0x41452e)
19:52:11   /usr/src/go/src/runtime/sigpanic_unix.go:26 (0x41ace4)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:281 (0x512166)
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:401 (0x43e715)
19:52:11   /usr/src/go/src/runtime/panic.go:387 (0x415388)
19:52:11   /usr/src/go/src/runtime/panic.go:42 (0x41452e)
19:52:11   /usr/src/go/src/runtime/sigpanic_unix.go:26 (0x41ace4)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:151 (0x5112bf)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:163 (0x5114c8)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:93 (0x710fd0)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:55 (0x710deb)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:49 (0x710d3c)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:188 (0x4ac31f)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:295 (0x513064)
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:2232 (0x440851)
19:52:11 ------------------------------
```

**Comment by jayunit100** [comment]
cc @gmarek @timothysc @ncdc .

**Comment by timothysc** [comment]
The fact that networking is marked as "flakey" is concerning, as I find it to be the single most important test.

**Comment by dchen1107** [comment]
@jayunit100 Actually multiple jenkins projects failed last night due to the same issue. I only mentioned above two (Empty, Services) because they are green last many days (> 10 days) without any flakiness. After reverting your PR, jenkins projects are back to green again. 

I poked several failed builds, and found right before those failures, e2e just finished a network test: Networking  should function for intrapod communication between all hosts in 3 parallel services. To me that network test should be marked as failure, but didn't. I guess this is the root cause here.

```
20:19:55 Networking 
20:19:55   should function for intrapod communication between all hosts in 3 parallel services [Conformance]
20:19:55   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:130
20:19:55 [BeforeEach] Networking
20:19:55   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:51
20:19:55 STEP: Creating a kubernetes client
20:19:55 >>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/config
20:19:55 STEP: Building a namespace api object
20:19:55 Oct 13 20:19:55.589: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-nettest-whusm
20:19:55 Oct 13 20:19:55.591: INFO: Get service account default in ns e2e-tests-nettest-whusm failed, ignoring for 2s: serviceaccounts "default" not found
20:19:57 Oct 13 20:19:57.594: INFO: Service account default in ns e2e-tests-nettest-whusm with secrets found. (2.005035574s)
20:19:57 STEP: Waiting for a default service account to be provisioned in namespace
20:19:57 Oct 13 20:19:57.594: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-nettest-whusm
20:19:57 Oct 13 20:19:57.596: INFO: Service account default in ns e2e-tests-nettest-whusm with secrets found. (2.04063ms)
20:19:57 [BeforeEach] Networking
20:19:57   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:52
20:19:57 STEP: Executing a successful http request from the external internet
20:19:57 [It] should function for intrapod communication between all hosts in 3 parallel services [Conformance]
20:19:57   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:130
20:19:57 Oct 13 20:19:57.688: INFO: running service test with timeout = 1m0s for 3
20:19:57 Oct 13 20:19:57.692: INFO: Node e2e-gce-minion-fr5s condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 20:15:46 -0700 PDT
20:19:57 Oct 13 20:19:57.692: INFO: Successfully found node e2e-gce-minion-fr5s readiness to be true
20:19:57 Oct 13 20:19:57.692: INFO: Node e2e-gce-minion-kqg1 condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 20:15:54 -0700 PDT
20:19:57 Oct 13 20:19:57.692: INFO: Successfully found node e2e-gce-minion-kqg1 readiness to be true
20:19:57 Oct 13 20:19:57.692: INFO: Node e2e-gce-minion-m6x5 condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 20:15:46 -0700 PDT
20:19:57 Oct 13 20:19:57.692: INFO: Successfully found node e2e-gce-minion-m6x5 readiness to be true
20:19:57 Oct 13 20:19:57.692: INFO: Waiting on ports to report back.  So far 0 have been discovered...
20:19:57 STEP: creating a service named "nettest-8080" in namespace "e2e-tests-nettest-whusm"
20:19:57 STEP: creating a service named "nettest-8081" in namespace "e2e-tests-nettest-whusm"
20:19:57 STEP: creating a service named "nettest-8082" in namespace "e2e-tests-nettest-whusm"
20:19:57 Oct 13 20:19:57.708: INFO: Created service successfully [nettest-8080]
20:19:57 Oct 13 20:19:57.708: INFO: launching pod per node.....
20:19:57 Oct 13 20:19:57.728: INFO: Created service successfully [nettest-8081]
20:19:57 Oct 13 20:19:57.728: INFO: launching pod per node.....
20:19:57 Oct 13 20:19:57.740: INFO: Created service successfully [nettest-8082]
20:19:57 Oct 13 20:19:57.740: INFO: launching pod per node.....
20:19:57 Oct 13 20:19:57.752: INFO: Created pod nettest-8080-rjcuu on node e2e-gce-minion-fr5s
20:19:57 Oct 13 20:19:57.767: INFO: Created pod nettest-8082-gq355 on node e2e-gce-minion-fr5s
20:19:57 Oct 13 20:19:57.770: INFO: Created pod nettest-8081-32gp3 on node e2e-gce-minion-fr5s
20:19:57 Oct 13 20:19:57.777: INFO: Created pod nettest-8080-qhl4t on node e2e-gce-minion-kqg1
20:19:57 Oct 13 20:19:57.795: INFO: Created pod nettest-8082-btoad on node e2e-gce-minion-kqg1
20:19:57 Oct 13 20:19:57.989: INFO: Created pod nettest-8081-zavsm on node e2e-gce-minion-kqg1
20:19:58 Oct 13 20:19:58.189: INFO: Created pod nettest-8080-hq7xo on node e2e-gce-minion-m6x5
20:19:58 Oct 13 20:19:58.189: INFO: Launched test pods for 8080
20:19:58 STEP: waiting for all of the webserver pods to transition to Running + reaching the Passing state.
20:19:58 Oct 13 20:19:58.189: INFO: Waiting up to 5m0s for pod nettest-8080-rjcuu status to be running
20:19:58 Oct 13 20:19:58.388: INFO: Created pod nettest-8082-szzb3 on node e2e-gce-minion-m6x5
20:19:58 Oct 13 20:19:58.388: INFO: Launched test pods for 8082
20:19:58 STEP: waiting for all of the webserver pods to transition to Running + reaching the Passing state.
20:19:58 Oct 13 20:19:58.388: INFO: Waiting up to 5m0s for pod nettest-8082-gq355 status to be running
20:19:58 Oct 13 20:19:58.588: INFO: Created pod nettest-8081-8ex92 on node e2e-gce-minion-m6x5
20:19:58 Oct 13 20:19:58.588: INFO: Launched test pods for 8081
20:19:58 STEP: waiting for all of the webserver pods to transition to Running + reaching the Passing state.
20:19:58 Oct 13 20:19:58.588: INFO: Waiting up to 5m0s for pod nettest-8081-32gp3 status to be running
20:19:58 Oct 13 20:19:58.785: INFO: Waiting for pod nettest-8080-rjcuu in namespace 'e2e-tests-nettest-whusm' status to be 'running'(found phase: "Pending", readiness: false) (595.93983ms elapsed)
20:19:58 Oct 13 20:19:58.985: INFO: Waiting for pod nettest-8082-gq355 in namespace 'e2e-tests-nettest-whusm' status to be 'running'(found phase: "Pending", readiness: false) (596.945089ms elapsed)
20:19:59 Oct 13 20:19:59.185: INFO: Waiting for pod nettest-8081-32gp3 in namespace 'e2e-tests-nettest-whusm' status to be 'running'(found phase: "Pending", readiness: false) (597.106175ms elapsed)
20:20:00 Oct 13 20:20:00.788: INFO: Waiting for pod nettest-8080-rjcuu in namespace 'e2e-tests-nettest-whusm' status to be 'running'(found phase: "Pending", readiness: false) (2.599210066s elapsed)
20:20:00 Oct 13 20:20:00.988: INFO: Waiting for pod nettest-8082-gq355 in namespace 'e2e-tests-nettest-whusm' status to be 'running'(found phase: "Pending", readiness: false) (2.600007209s elapsed)
20:20:01 Oct 13 20:20:01.189: INFO: Found pod 'nettest-8081-32gp3' on node 'e2e-gce-minion-fr5s'
20:20:01 STEP: waiting for connectivity to be verified [ port =  8081 ] 
20:20:01 Oct 13 20:20:01.189: INFO: Begin polling nettest-8081
20:20:02 Oct 13 20:20:02.792: INFO: Found pod 'nettest-8080-rjcuu' on node 'e2e-gce-minion-fr5s'
20:20:02 STEP: waiting for connectivity to be verified [ port =  8080 ] 
20:20:02 Oct 13 20:20:02.792: INFO: Begin polling nettest-8080
20:20:02 Oct 13 20:20:02.991: INFO: Found pod 'nettest-8082-gq355' on node 'e2e-gce-minion-fr5s'
20:20:02 STEP: waiting for connectivity to be verified [ port =  8082 ] 
20:20:02 Oct 13 20:20:02.991: INFO: Begin polling nettest-8082
20:20:06 Oct 13 20:20:06.795: INFO: Waiting up to 5m0s for pod nettest-8080-qhl4t status to be running
20:20:06 Oct 13 20:20:06.795: INFO: ... Another service has successfully been discovered: 0 ( nettest-8080 ) 
20:20:06 Oct 13 20:20:06.795: INFO: Waiting on ports to report back.  So far 1 have been discovered...
20:20:06 Oct 13 20:20:06.798: INFO: Found pod 'nettest-8080-qhl4t' on node 'e2e-gce-minion-kqg1'
20:20:06 STEP: waiting for connectivity to be verified [ port =  8080 ] 
20:20:06 Oct 13 20:20:06.798: INFO: Begin polling nettest-8080
20:20:06 Oct 13 20:20:06.995: INFO: Waiting up to 5m0s for pod nettest-8082-btoad status to be running
20:20:06 Oct 13 20:20:06.996: INFO: ... Another service has successfully been discovered: 1 ( nettest-8082 ) 
20:20:06 Oct 13 20:20:06.996: INFO: Waiting on ports to report back.  So far 2 have been discovered...
20:20:06 Oct 13 20:20:06.998: INFO: Found pod 'nettest-8082-btoad' on node 'e2e-gce-minion-kqg1'
20:20:06 STEP: waiting for connectivity to be verified [ port =  8082 ] 
20:20:06 Oct 13 20:20:06.998: INFO: Begin polling nettest-8082
20:20:07 Oct 13 20:20:07.193: INFO: Waiting up to 5m0s for pod nettest-8081-zavsm status to be running
20:20:07 Oct 13 20:20:07.193: INFO: ... Another service has successfully been discovered: 2 ( nettest-8081 ) 
20:20:07 Oct 13 20:20:07.193: INFO: Completed test on 3 port/services
20:20:07 [AfterEach] Networking
20:20:07   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:52
20:20:07 Oct 13 20:20:07.193: INFO: Waiting up to 1m0s for all nodes to be ready
20:20:07 Oct 13 20:20:07.197: INFO: Node e2e-gce-minion-fr5s condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 20:15:46 -0700 PDT
20:20:07 Oct 13 20:20:07.197: INFO: Successfully found node e2e-gce-minion-fr5s readiness to be true
20:20:07 Oct 13 20:20:07.197: INFO: Node e2e-gce-minion-kqg1 condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 20:15:54 -0700 PDT
20:20:07 Oct 13 20:20:07.197: INFO: Successfully found node e2e-gce-minion-kqg1 readiness to be true
20:20:07 Oct 13 20:20:07.197: INFO: Node e2e-gce-minion-m6x5 condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 20:15:46 -0700 PDT
20:20:07 Oct 13 20:20:07.197: INFO: Successfully found node e2e-gce-minion-m6x5 readiness to be true
20:20:07 STEP: Destroying namespace "e2e-tests-nettest-whusm" for this suite.
20:20:07 Oct 13 20:20:07.208: INFO: Found pod 'nettest-8081-zavsm' on node 'e2e-gce-minion-kqg1'
20:20:07 STEP: waiting for connectivity to be verified [ port =  8081 ] 
20:20:07 Oct 13 20:20:07.208: INFO: Begin polling nettest-8081
20:20:08 Oct 13 20:20:08.800: INFO: Unexpected response:
20:20:08 {
20:20:08   "kind": "Status",
20:20:08   "apiVersion": "v1",
20:20:08   "metadata": {},
20:20:08   "status": "Failure",
20:20:08   "message": "endpoints \"nettest-8080\" not found",
20:20:08   "reason": "NotFound",
20:20:08   "details": {
20:20:08     "name": "nettest-8080",
20:20:08     "kind": "endpoints"
20:20:08   },
20:20:08   "code": 404
20:20:08 }
20:20:08 
20:20:09 Oct 13 20:20:09.001: INFO: Unexpected response:
20:20:09 {
20:20:09   "kind": "Status",
20:20:09   "apiVersion": "v1",
20:20:09   "metadata": {},
20:20:09   "status": "Failure",
20:20:09   "message": "endpoints \"nettest-8082\" not found",
20:20:09   "reason": "NotFound",
20:20:09   "details": {
20:20:09     "name": "nettest-8082",
20:20:09     "kind": "endpoints"
20:20:09   },
20:20:09   "code": 404
20:20:09 }
20:20:09 
20:20:09 Oct 13 20:20:09.211: INFO: Unexpected response:
20:20:09 {
20:20:09   "kind": "Status",
20:20:09   "apiVersion": "v1",
20:20:09   "metadata": {},
20:20:09   "status": "Failure",
20:20:09   "message": "endpoints \"nettest-8081\" not found",
20:20:09   "reason": "NotFound",
20:20:09   "details": {
20:20:09     "name": "nettest-8081",
20:20:09     "kind": "endpoints"
20:20:09   },
20:20:09   "code": 404
20:20:09 }
20:20:09 
20:20:10 Oct 13 20:20:10.800: INFO: Unexpected response:
20:20:10 {
20:20:10   "kind": "Status",
20:20:10   "apiVersion": "v1",
20:20:10   "metadata": {},
20:20:10   "status": "Failure",
20:20:10   "message": "endpoints \"nettest-8080\" not found",
20:20:10   "reason": "NotFound",
20:20:10   "details": {
20:20:10     "name": "nettest-8080",
20:20:10     "kind": "endpoints"
20:20:10   },
20:20:10   "code": 404
20:20:10 }
20:20:10 
20:20:11 Oct 13 20:20:11.001: INFO: Unexpected response:
20:20:11 {
20:20:11   "kind": "Status",
20:20:11   "apiVersion": "v1",
20:20:11   "metadata": {},
20:20:11   "status": "Failure",
20:20:11   "message": "endpoints \"nettest-8082\" not found",
20:20:11   "reason": "NotFound",
20:20:11   "details": {
20:20:11     "name": "nettest-8082",
20:20:11     "kind": "endpoints"
20:20:11   },
20:20:11   "code": 404
20:20:11 }
20:20:11 
20:20:11 Oct 13 20:20:11.211: INFO: Unexpected response:
20:20:11 {
20:20:11   "kind": "Status",
20:20:11   "apiVersion": "v1",
20:20:11   "metadata": {},
20:20:11   "status": "Failure",
20:20:11   "message": "endpoints \"nettest-8081\" not found",
20:20:11   "reason": "NotFound",
20:20:11   "details": {
20:20:11     "name": "nettest-8081",
20:20:11     "kind": "endpoints"
20:20:11   },
20:20:11   "code": 404
20:20:11 }

20:20:12 • [SLOW TEST:16.630 seconds]
20:20:12 Networking
20:20:12 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:132
20:20:12   should function for intrapod communication between all hosts in 3 parallel services [Conformance]
20:20:12   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:130
```

Even I thinks which test failed is irrelevant now based on above issue, but below is the full output from failed e2e test: EmptyDir volumes should support (root,0666,default). 
```
19:52:04 ------------------------------
19:52:04 EmptyDir volumes 
19:52:04   should support (root,0666,default) [Conformance]
19:52:04   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/empty_dir.go:78
19:52:04 [BeforeEach] EmptyDir volumes
19:52:04   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:51
19:52:04 STEP: Creating a kubernetes client
19:52:04 >>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/config
19:52:04 STEP: Building a namespace api object
19:52:04 Oct 13 19:52:04.953: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-emptydir-6hypj
19:52:04 Oct 13 19:52:04.955: INFO: Get service account default in ns e2e-tests-emptydir-6hypj failed, ignoring for 2s: serviceaccounts "default" not found
19:52:05 STEP: Cleaning up the webserver pods
19:52:05 STEP: Cleaning up the service
19:52:06 Oct 13 19:52:06.958: INFO: Service account default in ns e2e-tests-emptydir-6hypj with secrets found. (2.004798904s)
19:52:06 STEP: Waiting for a default service account to be provisioned in namespace
19:52:06 Oct 13 19:52:06.958: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-emptydir-6hypj
19:52:06 Oct 13 19:52:06.960: INFO: Service account default in ns e2e-tests-emptydir-6hypj with secrets found. (1.920408ms)
19:52:06 [AfterEach] EmptyDir volumes
19:52:06   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:52
19:52:06 STEP: Collecting events from namespace "e2e-tests-emptydir-6hypj".
19:52:06 Oct 13 19:52:06.973: INFO: POD                                        NODE                 PHASE    GRACE  CONDITIONS
19:52:06 Oct 13 19:52:06.973: INFO: elasticsearch-logging-v1-06mn9             e2e-gce-minion-trb4  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:55 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: elasticsearch-logging-v1-y268x             e2e-gce-minion-7697  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:44 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: fluentd-elasticsearch-e2e-gce-minion-01eb  e2e-gce-minion-01eb  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:33 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: fluentd-elasticsearch-e2e-gce-minion-7697  e2e-gce-minion-7697  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:44 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: fluentd-elasticsearch-e2e-gce-minion-trb4  e2e-gce-minion-trb4  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:55 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: heapster-v10-qsh9e                         e2e-gce-minion-trb4  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:55 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: kibana-logging-v1-09tqp                    e2e-gce-minion-01eb  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:33 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: kube-dns-v9-gh5by                          e2e-gce-minion-7697  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:57 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: kube-ui-v3-0yohs                           e2e-gce-minion-7697  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:44 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: monitoring-influxdb-grafana-v2-kvqxa       e2e-gce-minion-01eb  Running   [{Ready True 0001-01-01 00:00:00 +0000 UTC 2015-10-13 19:46:33 -0700 PDT  }]
19:52:06 Oct 13 19:52:06.973: INFO: 
19:52:06 Oct 13 19:52:06.973: INFO: Waiting up to 1m0s for all nodes to be ready
19:52:06 Oct 13 19:52:06.976: INFO: Node e2e-gce-minion-01eb condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 19:46:33 -0700 PDT
19:52:06 Oct 13 19:52:06.976: INFO: Successfully found node e2e-gce-minion-01eb readiness to be true
19:52:06 Oct 13 19:52:06.976: INFO: Node e2e-gce-minion-7697 condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 19:46:44 -0700 PDT
19:52:06 Oct 13 19:52:06.976: INFO: Successfully found node e2e-gce-minion-7697 readiness to be true
19:52:06 Oct 13 19:52:06.976: INFO: Node e2e-gce-minion-trb4 condition 1/1: type: Ready, status: True, reason: "KubeletReady", message: "kubelet is posting ready status", last transition time: 2015-10-13 19:46:54 -0700 PDT
19:52:06 Oct 13 19:52:06.976: INFO: Successfully found node e2e-gce-minion-trb4 readiness to be true
19:52:06 STEP: Destroying namespace "e2e-tests-emptydir-6hypj" for this suite.
19:52:11 
19:52:11 •! Panic in Spec Setup (BeforeEach) [7.044 seconds]
19:52:11 EmptyDir volumes
19:52:11 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/empty_dir.go:95
19:52:11   should support (root,0666,default) [Conformance] [BeforeEach]
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/empty_dir.go:78
19:52:11 
19:52:11   Test Panicked
19:52:11   runtime error: invalid memory address or nil pointer dereference
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:401
19:52:11 
19:52:11   Full Stack Trace
19:52:11   /usr/src/go/src/runtime/panic.go:42 (0x41452e)
19:52:11   /usr/src/go/src/runtime/sigpanic_unix.go:26 (0x41ace4)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:270 (0x511e50)
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:401 (0x43e715)
19:52:11   /usr/src/go/src/runtime/panic.go:387 (0x415388)
19:52:11   /usr/src/go/src/runtime/panic.go:42 (0x41452e)
19:52:11   /usr/src/go/src/runtime/sigpanic_unix.go:26 (0x41ace4)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:281 (0x512166)
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:401 (0x43e715)
19:52:11   /usr/src/go/src/runtime/panic.go:387 (0x415388)
19:52:11   /usr/src/go/src/runtime/panic.go:42 (0x41452e)
19:52:11   /usr/src/go/src/runtime/sigpanic_unix.go:26 (0x41ace4)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:151 (0x5112bf)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:163 (0x5114c8)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:93 (0x710fd0)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:55 (0x710deb)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:49 (0x710d3c)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:188 (0x4ac31f)
19:52:11   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/networking.go:295 (0x513064)
19:52:11   /usr/src/go/src/runtime/asm_amd64.s:2232 (0x440851)
```

**Comment by ncdc** [comment]
I've discovered another flake and it has to do with the order that things get deleted:

```
STEP: Destroying namespace "e2e-tests-nettest-heo8y" for this suite.
STEP: Cleaning up the service

• Failure in Spec Teardown (AfterEach) [15.431 seconds]
Networking
/home/vagrant/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/networking.go:132
  should function for intrapod communication between all hosts in 3 parallel services [Conformance] [AfterEach]
  /home/vagrant/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/networking.go:130

  Oct 14 13:38:31.225: unable to delete svc nettest-8080: services "nettest-8080" not found

  /home/vagrant/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/networking.go:271
```

**Comment by jayunit100** [comment]
Interesting, looks like youv'e reproduced this in vagrant.  I've run these in vagrant before (without issue)... but let me try again.

**Comment by ncdc** [comment]
I haven't reproduced the nil pointer yet

**Comment by jayunit100** [comment]
Yup, nil pointer seems to be a GCE specific thing? maybe based on url or something...

**Comment by ncdc** [comment]
My guess is that your Wait.Poll code is still running in a goroutine when the framework goes out of scope.

**Comment by jayunit100** [comment]
Yes  I think you are right ; definitely some variables are being referenced after the test has already "failed".

**Comment by jayunit100** [comment]
Ok, after playing with a 3 node vagrant cluster, I was able to recreate this.  
- It seems like some funniness around how the AfterEach/BeforeEach are triggered,  and it only happens in clusters, which can lead to the polling continuing. 
- it seems to be a recent change (a month ago, i ran this in a cluster, and it didn't have problems).

The mechanism is less important, the most important thing is that when this is reissued we have a rock-solid way to review-before-merge.  Because, networking.go is **not** run in the pull-requests CI.  I've created an issue for that separately.

1) Run it with 3 VMs , not two, or one.
2) Run it multiple times.

This is of course until #15656 .
